misago/legal/views.py

```diff
@@ -1,69 +1,33 @@
+from django.http import HttpResponse
+from django.shortcuts import render
 
-from misago.conf import settings
-from misago.core.cache import cache
+from misago.legal.models import LegalAgreement
 from misago.markup import common_flavour
 
+def legals_active(request):
+    legal_active = LegalAgreement.objects.filter(is_active=True).all()
+    response = HttpResponse(" \n".join(str(v) for v in list(legal_active)))
+    response['Content-Type'] = 'text/plain'
 
-def get_parsed_content(request, setting_name):
-    cache_name = 'misago_legal_%s' % setting_name
-    cached_content = cache.get(cache_name)
-
-    unparsed_content = settings.get_lazy_setting(setting_name)
-
-    checksum_source = force_bytes('%s:%s' % (unparsed_content, settings.SECRET_KEY))
-    unparsed_checksum = md5(checksum_source).hexdigest()
-
-    if cached_content and cached_content.get('checksum') == unparsed_checksum:
-        return cached_content['parsed']
-    else:
-        parsed = common_flavour(request, None, unparsed_content)['parsed_text']
-        cached_content = {
-            'checksum': unparsed_checksum,
-            'parsed': parsed,
-        }
-        cache.set(cache_name, cached_content)
-        return cached_content['parsed']
-
-
-def privacy_policy(request):
-    if not (settings.privacy_policy or settings.privacy_policy_link):
-        raise Http404()
-
-    if settings.privacy_policy_link:
-        return redirect(settings.privacy_policy_link)
-
-    parsed_content = get_parsed_content(request, 'privacy_policy')
-
-    return render(
-        request, 'misago/privacy_policy.html', {
-            'id': 'privacy-policy',
-            'title': settings.privacy_policy_title or _("Privacy policy"),
-            'link': settings.privacy_policy_link,
-            'body': parsed_content,
-        }
-    )
-
-
-def terms_of_service(request):
-    if not (settings.terms_of_service or settings.terms_of_service_link):
-        raise Http404()
+    return response
 
-    if settings.terms_of_service_link:
-        return redirect(settings.terms_of_service_link)
+def legal_title(request, title):
+    """
+    This function returns the active specified legal type
+    :param title: Legal Type name.
+    :return: legal.text
+    """
+    try:
+        legal = LegalAgreement.objects.filter(is_active=True).get(title=title)
+    except LegalAgreement.DoesNotExist:
+        raise Http404
 
-    parsed_content = get_parsed_content(request, 'terms_of_service')
+    parsed = common_flavour(request, None, legal.text)
 
     return render(
-        request, 'misago/terms_of_service.html', {
-            'id': 'terms-of-service',
-            'title': settings.terms_of_service_title or _("Terms of service"),
-            'link': settings.terms_of_service_link,
-            'body': parsed_content,
+        request, 'misago/legal.html', {
+            'id': legal.title,
+            'title': dict(legal.LEGAL_CHOICES)[legal.title],
```

Comment:
How can, e.g. polish person, customize the legal agreements title? Misago features must be i18n out of the box.